openvino/conditional_compilation/CMakeLists.txt

```diff
@@ -36,6 +36,11 @@ elseif(SELECTIVE_BUILD STREQUAL "ON")
 
     target_compile_definitions(${TARGET_NAME} INTERFACE SELECTIVE_BUILD)
 
+    if (UNIX)
+        # After disabling a block of code, some variables might be unused.
+        target_compile_options(${TARGET_NAME} INTERFACE -Wno-unused-variable)
```

Comment:
Did you get unused variables for the CPU plugin? Can we avoid it?

Reply:
I got unused variables for the oneDNN library. I think we can analyse and refactor all that places.

Reply:
In this case I propose to disable these warnings only for oneDNN project. Because in this case we don't hide warnings for other components.

Reply:
In case of conditional compilation we can have a lot of cases when variables are not used. I think it's ok to disable such warnings. 

And it's not related to oneDNN plugin. 

BTW, suggest adding `-Wunused-function` as well.

Reply:
I asked to don't disable these warnings for whole OpenVINO because for example in nGraph we don't have such issues.
And if we faced with unused variable in the majority situations (for nGraph) it means that we can disable more code than we already disabled.

Reply:
Current changes disable such warnings only for libraries where conditional compilation is used, not for the whole OpenVINO project :)

Reply:
The set of libraries, where conditional compilation is used, contains the nGraph library. 😄

Reply:
Agreed to keep globally only `-Wunused-function -Wunused-parameter`

Reply:
Unused options list is wider. Can we add some other options form this list?
Wno-unused
Wno-unused-but-set-parameter
Wno-unused-but-set-variable
Wno-unused-const-variable
Wno-unused-function
Wno-unused-label
Wno-unused-local-typedefs
Wno-unused-parameter
Wno-unused-result
Wno-unused-value

Reply:
Looks like we can also use `-Wunused-local-typedefs`, not sure about others..

Reply:
added